ayon_api/entity_hub.py

```diff
@@ -1485,6 +1486,704 @@ def fill_children_ids(self, children_ids):
+            or self._original_short_name != self._short_name
+            or self._original_index != self._index
+            or self._original_state != self._state
+            or self._original_icon != self._icon
+            or self._original_color != self._color
+        )
+
+    def delete(self):
+        """Remove status from project statuses object."""
+
+        if self._project_statuses is not None:
+            self._project_statuses.remove(self)
+
+    def get_index(self):
+        """Get index of status.
+
+        Returns:
+            Union[int, None]: Index of status or None if status is not under
+                project.
+        """
+
+        return self._index
+
+    def set_index(self, index, **kwargs):
+        """Change status index.
+
+        Returns:
+            Union[int, None]: Index of status or None if status is not under
+                project.
+        """
+
+        if kwargs.get("from_parent"):
+            self._index = index
+        else:
+            self._project_statuses.set_status_index(self, index)
+
+    def get_name(self):
+        """Status name.
+
+        Returns:
+            str: Status name.
+        """
+
+        return self._name
+
+    def set_name(self, name):
+        """Change status name.
+
+        Args:
+            name (str): New status name.
+        """
+
+        if not isinstance(name, six.string_types):
+            raise TypeError("Name must be a string.")
+        if name == self._name:
+            return
+        self._name = name
+        self._slugified_name = None
+
+    def get_short_name(self):
+        """Status short name 3 letters tops.
+
+        Returns:
+            str: Status short name.
+        """
+
+        return self._short_name
+
+    def set_short_name(self, short_name):
+        """Change status short name.
+
+        Args:
+            short_name (str): New status short name. 3 letters tops.
+        """
+
+        if not isinstance(short_name, six.string_types):
+            raise TypeError("Short name must be a string.")
+        self._short_name = short_name
+
+    def get_icon(self):
+        """Name of icon to use for status.
+
+        Returns:
+            str: Name of the icon.
+        """
+
+        return self._icon
+
+    def set_icon(self, icon):
+        """Change status icon name.
+
+        Args:
+            icon (str): Name of the icon.
+        """
+
+        if icon is None:
+            icon = ""
+        if not isinstance(icon, six.string_types):
+            raise TypeError("Icon name must be a string.")
+        self._icon = icon
+
+    @property
+    def slugified_name(self):
+        """Slugified and lowere status name.
+
+        Can be used for comparison of existing statuses. e.g. 'In Progress'
+            vs. 'in-progress'.
+
+        Returns:
+            str: Slugified and lower status name.
+        """
+
+        if self._slugified_name is None:
+            self._slugified_name = self.slugify_name(self.name)
```

Comment:
Why not call `slugify_string(self.name.lower())` here directly?

Reply:
To have single place of the name slugify, so it's 100% same as public function logic (`slufigy_name`).